### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by Kitio-Tek**
This pull request includes the initial setup and implementation of a Flask-based web application for the "Xmas of Things" project. 

Application setup and main logic:

Added the main Flask application logic, including routes for the index page and sensor data endpoint, TTN listener setup, and action command handling.

Web interface: app.py
main dashboard, including sections for displaying sensor data and sending action commands.

Helper functions:

ttn_client.py
Added a helper function to send downlink actions to TTN.

Note: You may delete files: read_actions_queue.py, read_sensors.py , and start_logic_map.py, no need to have them, they are superflu

### CODE REVIEW

File: API/src/app.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    ActionsTTNPayload,
+    ActionsEnum,
+    sensors_topic,
+    actions_topic,
+)
+from pyxmas.settings import TTN_APP_ID, TTN_API_KEY, TTN_BASE_URL, TTN_PORT
+from datetime import datetime, timezone
+
+# Ensure pyxmas is in the Python path
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'pyxmas')))
```
Comment (lbo462) [inline API/src/app.py:21]: Why ? You're in the same directory

File: API/src/app.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'your_secret_key'  # Replace with a secure key in production
```
Comment (lbo462) [inline API/src/app.py:24]: You have pyxmas.settings implementing such thing, take a look.

File: API/src/app.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Global variable to store the latest sensor data
+latest_sensor_data = {
```
Comment (lbo462) [inline API/src/app.py:31]: The logic map already has this. Check `logic_map.last_sensors_data`

File: API/src/app.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    'brightness': 'N/A',
+    'loudness': 'N/A',
+    'temperature': 'N/A',
+    'timestamp': 'N/A'
+}
+data_lock = threading.Lock()
+
+def on_sensors_data(sensors_data: SensorsTTNPayload):
+    global latest_sensor_data
```
Comment (lbo462) [inline API/src/app.py:40]: From previous comment: logic_map already has this, and you should avoid using the keyword `global` with python

File: API/src/app.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    with data_lock:
+        latest_sensor_data = {
+            'brightness': sensors_data.brightness,
+            'loudness': sensors_data.loudness,
+            'temperature': sensors_data.temperature,
+            'timestamp': datetime.now(timezone.utc).strftime('%Y-%m-%d %H:%M:%S UTC')
+        }
+    logger.info(f"Received sensor data: {latest_sensor_data}")
+
+def start_ttn_listener():
```
Comment (lbo462) [inline API/src/app.py:50]: You're reinventing the wheel here, why don't just call the logic_map instance ?

File: API/src/ttn_client.py
Comment (lbo462) [inline API/src/ttn_client.py:1]: Why do you rewrite this ?
Why don't use the already existing ttn_al and the logic map, both featuring this `send_downlink` function ?

File: API/src/app.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            action_id = int(action_id)
+            if action_id not in ActionsEnum._value2member_map_:
+                raise ValueError("Invalid action ID.")
+            action_enum = ActionsEnum(action_id)
+        except (ValueError, TypeError) as e:
+            status_message = f"Error: {e}"
+            status_type = "danger"
+            logger.error(f"Action ID validation error: {e}")
+            # Proceed to render the page with the error message
+            sensor_data = get_latest_sensor_data()
```
Comment (lbo462) [inline API/src/app.py:93]: As said previously, use the logic map for getting the sensors data

**Review by lbo462 — CHANGES_REQUESTED** [review]
Good-looking Flask app, but do not rewrite existing code.
Take a look at the logic_map as it's used in the `start_logic_map.py` script.
Plus, you absolutely need to keep track of the village state, also using what's shown in `main.py` ! Otherwise, we'll get weird behaviors with flashing light.

Currently, you're not even starting the logic map, thus, all previous work is thrown to trash ...

File: API/src/app.py (hunk 6 above)
Comment (Kitio-Tek) [reply API/src/app.py]: Changes done on app.py

**Pull request closed without merging**